### PR TITLE
Chore/file refactor

### PR DESCRIPTION
**Pull request by SHolleworth**
Moved workspace and client into a single src/ directory.
Refactored workspace to share main services, removed most workspace specific services.
Applied shared formatting and linting rules to workspace.

### CODE REVIEW

File: src/src/client/App/LiveRates/Tile/__tests__/Tile.rfq.test.tsx [hunk 1]
```diff
@@ -189,7 +190,6 @@ describe("Tile/rfq", () => {
     const request: Partial<ExecutionRequest> = {
       ...originalRequest,
     }
-    delete request.id
```
Comment (SHolleworth) [inline src/src/client/App/LiveRates/Tile/__tests__/Tile.rfq.test.tsx:192]: id removed from the ExecutionRequest type

File: src/src/client/OpenFin/apps/Launcher/NlpSuggestions/executions/TradeExecution/TradeExecutionWorkflow.tsx [hunk 2]
```diff
@@ -58,7 +58,7 @@ const SuccessContent: React.FC<ExecutionTrade> = (trade) => {
   return (
     <>
       <p>
-        <strong>Trade ID: {trade.id}</strong>
+        <strong>Trade ID: {trade.tradeId}</strong>
```
Comment (SHolleworth) [inline src/src/client/OpenFin/apps/Launcher/NlpSuggestions/executions/TradeExecution/TradeExecutionWorkflow.tsx:61]: replace what was a locally incrementing id with the proper one from the backend

File: src/src/services/currencyPairs/currencyPairs.ts [hunk 3]
```diff
@@ -36,3 +36,7 @@ export const [useCurrencyPair, getCurrencyPair$] = bind((symbol: string) =>
     distinctUntilChanged(),
   ),
 )
+
+export const currencyPairSymbols$ = currencyPairs$.pipe(
```
Comment (SHolleworth) [inline src/src/services/currencyPairs/currencyPairs.ts:40]: small helper service

File: src/src/services/executions/executions.ts [hunk 4]
```diff
@@ -29,25 +29,22 @@ const mapExecutionToPayload = (e: ExecutionRequest): ExecuteTradeRequest => {
...
-      currencyPair: trade.currencyPair,
-      dealtCurrency: trade.dealtCurrency,
-      direction: trade.direction,
-      notional: trade.notional,
-      spotRate: trade.spotRate,
-      status: ExecutionStatus[trade.status],
-      tradeId: Number(trade.tradeId), // TODO: talk with hydra team
-      tradeDate: new Date(),
-      valueDate: new Date(trade.valueDate),
-      id,
```
Comment (SHolleworth) [inline src/src/services/executions/executions.ts:48]: remoivng unused

File: src/src/services/executions/executions.ts [hunk 5]
```diff
@@ -29,25 +29,22 @@ const mapExecutionToPayload = (e: ExecutionRequest): ExecuteTradeRequest => {
   }
 }
 
-const mapResponseToTrade =
-  (id: string) =>
-  ({ trade }: ExecutionResponse): ExecutionTrade => {
```
Comment (SHolleworth) [inline src/src/services/executions/executions.ts:34]: it was uneccessary to map the trades to ids

File: src/src/services/prices/prices.service-mock.ts [hunk 6]
```diff
@@ -60,15 +62,23 @@ export const [, getIsSymbolDataStale$] = bind(
...
+      (acc, [price, ccyPair]) => ({
         ...price,
         movementType:
           acc === undefined
             ? PriceMovementType.NONE
             : price.mid > acc.mid
             ? PriceMovementType.UP
             : PriceMovementType.DOWN,
+
+        spread: calculateSpread(
```
Comment (SHolleworth) [inline src/src/services/prices/prices.service-mock.ts:76]: calculating spread here to make it available across the app

File: src/src/services/prices/prices.ts [hunk 7]
```diff
@@ -54,15 +71,24 @@ export const [usePrice, getPrice$] = bind((symbol: string) =>
...
         ...price,
+
         movementType:
           acc === undefined
             ? PriceMovementType.NONE
             : price.mid > acc.mid
             ? PriceMovementType.UP
             : PriceMovementType.DOWN,
+
+        spread: calculateSpread(
```
Comment (SHolleworth) [inline src/src/services/prices/prices.ts:86]: calculating spread here to make it available across the app

File: src/src/workspace/home/index.ts [hunk 8]
```diff
@@ -76,105 +76,17 @@ export async function registerHome(): Promise<HomeRegistration> {
         lastResponse.revoke(result.key)
       }
     } else {
-      let pageToLaunch = await getPage(result.data.pageId)
+      const pageToLaunch = await getPage(result.data.pageId)
       await launchPage(pageToLaunch)
     }
   }
 
-  const handleAppSelection = async (appEntry: App) => {
```
Comment (SHolleworth) [inline src/src/workspace/home/index.ts:84]: moved into utils

File: src/src/workspace/home/nlpProvider.ts [hunk 9]
```diff
@@ -137,13 +153,19 @@ const getBaseSpotTemplate = (price: Price): TemplateFragment => {
             createImage("movementUp", "Price Movement Up", {
               width: "10px",
               height: "10px",
-              visibility: price.lastMove === "up" ? "visible" : "hidden",
+              visibility:
+                price.movementType === PriceMovementType.UP
```
Comment (SHolleworth) [inline src/src/workspace/home/nlpProvider.ts:157]: using the client Price type

File: src/src/workspace/home/nlpProvider.ts [hunk 10]
```diff
@@ -432,9 +461,27 @@ const constructTradeExecutedTemplateContent = (trade: Trade) => {
...
+  let data
+
+  switch (trade.status) {
+    case ExecutionStatus.Rejected:
+      data = {
+        tradeId: `Trade ID: ${trade.tradeId.toString()}`,
+        rejectedLabel: "Your trade has been rejected",
+      }
+      break
+    case ExecutionStatus.CreditExceeded:
```
Comment (SHolleworth) [inline src/src/workspace/home/nlpProvider.ts:473]: Using the client execution$ granted us more specific fail states

File: src/src/workspace/provider.ts [hunk 11]
```diff
@@ -40,13 +43,16 @@ async function init() {
   await registerDock()
   await showHome()
 
+  const sub = registerSimulatedDealerResponses()
```
Comment (SHolleworth) [inline src/src/workspace/provider.ts:46]: registering here so we can receive quotes after creating an rfq with nlp

File: src/tsconfig.json [hunk 12]
```diff
@@ -2,7 +2,8 @@
   "compilerOptions": {
     "baseUrl": "./src",
     "paths": {
-      "@/*": ["./*"]
+      "@/*": ["./*"],
+      "~/*": ["../*"]
```
Comment (SHolleworth) [inline src/tsconfig.json:6]: added aroot alias

File: src/src/client/App/LiveRates/Tile/PriceMovement/PriceMovement.tsx [hunk 13]
```diff
@@ -123,18 +105,11 @@ const PriceFromQuote = ({
...
-  const spread = calculateSpread(ask, bid, ratePrecision, pipsPosition)
-  return (
-    <PriceMovementStyle isAnalyticsView={isAnalytics}>
-      <MovementValue>{spread}</MovementValue>
-    </PriceMovementStyle>
-  )
-}
+}) => (
+  <PriceMovementStyle isAnalyticsView={isAnalytics}>
+    <MovementValue>{rfqResponse.price.spread}</MovementValue>
```
[the remark this answers is not in the document]
Comment (TobyLi96) [reply src/src/client/App/LiveRates/Tile/PriceMovement/PriceMovement.tsx:110]: Is this saying the spread is now calculated in the BE?

File: src/src/client/App/Analytics/AnalyticsCore.tsx [hunk 14]
```diff
@@ -1,11 +1,11 @@
 import { Subscribe } from "@react-rxjs/core"
 import { merge } from "rxjs"
 
-import { supportsTearOut } from "@/App/TearOutSection/supportsTearOut"
-import { TearOutComponent } from "@/App/TearOutSection/TearOutComponent"
+import { supportsTearOut } from "@/client/App/TearOutSection/supportsTearOut"
```
[the remark this answers is not in the document]
Comment (TobyLi96) [reply src/src/client/App/Analytics/AnalyticsCore.tsx:4]: Could definitely use one for at least the client folder

File: src/src/client/App/LiveRates/Tile/PriceMovement/PriceMovement.tsx (hunk 13 above)
[the remark this answers is not in the document]
Comment (TobyLi96) [reply src/src/client/App/LiveRates/Tile/PriceMovement/PriceMovement.tsx:110]: Never mind, saw your comment below.

**Review by TobyLi96 — COMMENTED** [review]
Looks good. Although the `src/src` file structure is a bit weird to me. Is it maybe better to flatten the first src directory?

**Comment by SHolleworth** [comment]
> Looks good. Although the `src/src` file structure is a bit weird to me. Is it maybe better to flatten the first src directory?

Agreed

**Review by TobyLi96 — APPROVED** [review]

**Pull request merged**